pyheaders/__init__.py

```diff
@@ -55,45 +57,51 @@ def _load_file(filename: _Path, /, extra_args: _Iterable[_Text] = None, *, verbo
+              verbose: bool = False, initial_scope: cpp.Scope = None,
+              clang_path: _Path = None, commands_parser: compiler.CommandsParser = None,
+              excludes: _List = None, **run_plugin_kwargs) -> SrcData:
     '''
     Load all constants from ``path`` (a ``str`` or ``bytes`` instance containing a
     path to a file or a directory containing C++ code) to a Python object.
 
-    @param path         The path of the file or directory to load. If a directory is
+    @param paths        The paths of the files/directories to load. If a directory is
                         given, all files whose extension is recognized as a C/C++ source
-                        code extension are loaded.
+                        code extension are loaded. A glob path is accepted as well.
     @param extra_args   Additional compilation arguments on top of the compile commands.
     @param verbose      If ``True`` is provided, don't suppress compiler errors.
     @param initial_scope The initial scope to use, defaults to a new empty scope.
     @param clang_path   The full path of the clang executable.
     @param commands_parser The CommandsParser object the compiler should use.
     @param run_plugin_kwargs Additional args for run_plugin().
+    @param excludes     List of glob paths to exclude when searching within a directory.
 
-    @returns Scope
+    @returns SrcData
     '''
-    if os.path.isfile(path):
-        return _load_file(path, extra_args=extra_args, verbose=verbose, initial_scope=initial_scope, **run_plugin_kwargs)
-
-    if os.path.isdir(path):
-        returned_data = SrcData()
-        if initial_scope is not None:
-            returned_data.scope = initial_scope
-
-        source_files = (os.path.join(dirpath, filename) for dirpath, _, files in os.walk(path) for filename in files
-                        if os.path.splitext(filename)[-1] in compiler.CPP_SOURCE_FILES_EXTENSIONS)
-        for filename in source_files:
-            filename = os.path.join(path, filename)
-
-            returned_data.update(_load_file(filename, extra_args=extra_args, verbose=verbose,
-                                            initial_scope=returned_data.scope, exec_path=clang_path,
-                                            commands_parser=commands_parser, **run_plugin_kwargs))
-
-        return returned_data
-
-    raise ValueError('path is neither a file nor a directory.')
+    returned_data = SrcData()
+    if initial_scope is not None:
+        returned_data.scope = initial_scope
+
+    source_files = set()
+    for path in paths:
+        if os.path.isfile(path):
+            source_files.add(path)
+        elif os.path.isdir(path):
+            source_files |= set(os.path.join(dirpath, filename) for dirpath, _, files in os.walk(path) for filename in files
+                                if os.path.splitext(filename)[-1] in compiler.CPP_SOURCE_FILES_EXTENSIONS)
```

Comment:
Set comprehensions ftw!
```suggestion
            source_files |= {os.path.join(dirpath, filename) for dirpath, _, files in os.walk(path) for filename in files
                                if os.path.splitext(filename)[-1] in compiler.CPP_SOURCE_FILES_EXTENSIONS}
```